packages/session-replay-browser/src/session-replay.ts

```diff
@@ -205,9 +205,17 @@ export class SessionReplay implements AmplitudeSessionReplay {
       if (this.config.debugMode) {
         eventProperties[SESSION_REPLAY_DEBUG_PROPERTY] = this.getSessionReplayDebugPropertyValue();
       }
+      void this.addCustomRRWebEvent('get-sr-props', {
+        shouldRecord,
+        eventProperties: eventProperties,
+      });
```

Comment:
The only thing I'm wondering about here is if this should also check the storage size estimate every time. It is async, but I wonder if there's a lot of overhead for checking that on every Amplitude event.

Reply:
Hmm I'm conflicted, we don't necessarily need it for every event, but if we don't include it in events we may only capture it at the beginning of the recording and miss out on valuable data from further in the recording. From a quick google search I don't see any notes about performance impact from this method. We are not blocking the execution of the event on this (hence the void), so it will allow the event to process through as normal. The other option would be to throttle it, to send it say only every 10 events or so, wdyt?

Reply:
>The other option would be to throttle it, to send it say only every 10 events or so, wdyt?

Yeah I think that works. An alternative would be if it's possible to determine at key points, but I think the sampling should be fine.